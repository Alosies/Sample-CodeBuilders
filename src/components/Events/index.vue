<template>
  <div>
    <!-- heading -->
    <div class="heading">
      <div style="display: flex">
        <v-icon size="3rem">
          event_available
        </v-icon>
        <h1 class="primary--text">
          Events
        </h1>
      </div>
      <v-btn @click="showModal = true"
             class="addNewEvent"
             color="primary">Add Event</v-btn>
    </div>
    <!-- list of events -->
    <div v-if="allEvents">
      <template v-for="(event, index) in allEvents">
        <v-card :key="index">
          <v-card-title primary-title
                        class="event">
            <div>
              <h3 class="headline mb-0">
                {{ event.title }}
              </h3>
              <div>{{ event.description }}</div>
            </div>
            <div>
              <div class="headline primary--text">
                {{ event.schedule.date}}
              </div>
              <div class="subheading">
                {{ event.schedule.month}}
              </div>
            </div>
          </v-card-title>
        </v-card>
      </template>
    </div>

    <!-- Modal for adding events -->
    <v-dialog v-model="showModal"
              width="600">
      <add-event @closeModal="showModal = false" />
    </v-dialog>
  </div>
</template>

<script>
import { mapGetters } from 'vuex'
import AddEvent from './ChildComponents/AddEvent'

export default {
  components: {
    AddEvent
  },
  data() {
    return {
      showModal: false
    }
  },
  computed: {
    ...mapGetters(['allEvents'])
  },
  created() {
    this.$store.dispatch('fetchEvents')
  }
}
</script>

<style lang="stylus" scoped>
.heading, .event {
  display: grid;
  align-items: center;
  grid-auto-flow: column;
  justify-content: space-between;
}

// .event {
// display: grid;
// grid-auto-flow: column;
// justify-content: space-between;
// }
h1 {
  padding: 1rem;
}
</style>
