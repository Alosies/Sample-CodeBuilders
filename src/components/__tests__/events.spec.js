import { shallowMount } from '@vue/test-utils'
import Events from '@/components/Events'
import Vuetify from 'vuetify'
import Vue from 'vue'

Vue.use(Vuetify)

// Setup - mock Store
const mockStore = {
  dispatch: jest.fn(),
  getters: { allEvents: [] }
}

const mocks = {
  $store: mockStore
}

describe('BlogPosts/index.vue', () => {
  let build
  beforeEach(() => {
    build = () => {
      const wrapper = shallowMount(Events, { mocks })
      return {
        wrapper
      }
    }
  })

  it('Events Component renders', () => {
    // Act - mount the component
    const { wrapper } = build()
    // Assert - Check the Snapshot
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Dispatches action to the store to fetches Events', () => {
    // Assert - Check whether the correct action was disptached
    expect(mockStore.dispatch).toHaveBeenCalledWith('fetchEvents')
  })

  it('Fetches allPosts from Store', () => {
    // Act - mount the component
    const { wrapper } = build()
    // Assert - Check the computed property
    expect(wrapper.vm.allEvents).toBe(mockStore.getters.allEvents)
  })
})
