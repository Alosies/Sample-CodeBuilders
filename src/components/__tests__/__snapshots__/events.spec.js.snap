// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogPosts/index.vue Events Component renders 1`] = `
<div>
  <div class="heading">
    <div style="display: flex;">
      <v-icon-stub size="3rem">
        event_available
      </v-icon-stub>
      <h1 class="primary--text">
        Events
      </h1>
    </div>
    <v-btn-stub color="primary" activeclass="v-btn--active" tag="button" type="button" class="addNewEvent">Add Event</v-btn-stub>
  </div>
  <div></div>
  <v-dialog-stub contentclass="" maxwidth="none" origin="center center" width="600" transition="dialog-transition">
    <add-event-stub></add-event-stub>
  </v-dialog-stub>
</div>
`;
